pkg/kubelet/cm/cpumanager/policy_static_test.go

```diff
@@ -514,10 +558,18 @@ func TestStaticPolicyAdd(t *testing.T) {
 	for _, testCase := range smtalignOptionTestCases {
 		runStaticPolicyTestCase(t, testCase)
 	}
+	for _, testCase := range alignBySocketOptionTestCases {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, pkgfeatures.CPUManagerPolicyAlphaOptions, true)()
```

Comment:
It's odd to put the defer in the middle of a for loop like this. Defers are function scoped and will only be run once this entire function exits. You are essentially adding a new function to run at function exit each time around the loop. Is this your intention here?

Reply:
You will likely need to add a wrapping function around `runStaticPolicyTestCase()` called something like `runStaticPolicyTestCaseWithFeatureGates()` that sets up the necessary feature gates around the test.

Reply:
fixed in latest commit